File: 卡码/代码随想录/1/09-动态规划/Python/05-子序列问题/02-子序列(连续)/LeetCode_0053.py
#53. 最大子数组和
from typing import *

class Solution:
  def maxSubArray(self, nums: List[int]) -> int:
    res:int=nums[0]
    dp:List[int]=[0]*(len(nums))
    dp[0]=nums[0]
    for i in range(1,len(nums)):
      dp[i]=max(dp[i-1]+nums[i],nums[i])
      res=dp[i] if dp[i]>res else res
    return res

File: top100/Java/top100/src/main/java/com/kurumi/leetCode/top100/普通数组/LeetCode_0053.java
package com.kurumi.leetCode.top100.普通数组;

/**
 * @Author kurumi
 * @Date 2026-06-30 下午5:53:29
 * @Description 53. 最大子数组和
 */
public class LeetCode_0053 {
    static class Solution {
        public int maxSubArray(int[] nums) {
            int cur = nums[0], res = nums[0];
            for (int i = 1; i < nums.length; i++) {
                cur = Math.max(cur + nums[i], nums[i]);
                res = Math.max(res, cur);
            }
            return res;
        }
    }
}
